.circleci/config.yml

```diff
@@ -168,8 +170,304 @@ jobs:
+      - save_cache:
+          key: v1-yarn-install-{{ checksum "yarn.lock" }}
+          paths: 
+            - node_modules
+            - packages/common-ts/node_modules
+            - packages/contracts/node_modules
+            - packages/core-utils/node_modules
+            - packages/data-transport-layer/node_modules
+            - packages/message-relayer/node_modules
+            - packages/replica-healthcheck/node_modules
+            - packages/sdk/node_modules
+            - integration-tests/node_modules
+      - run:
+          name: Build monorepo
+          command: yarn build
+      - save_cache:
+          key: v1-yarn-build-{{ .Revision }}
+          paths:
+            - "."
+
+  contracts-slither:
+    docker:
+      - image: mslipper/slither:latest
+    steps:
+      - restore_cache:
+          keys:
+            - v1-yarn-build-{{ .Revision }}
+      - checkout
+      - run:
+          name: Run Slither
+          command: yarn test:slither
+          working_directory: packages/contracts
+
+  contracts-tests:
+    docker:
+      - image: mslipper/slither:latest
+    resource_class: xlarge
+    steps:
+      - restore_cache:
+          keys:
+            - v1-yarn-build-{{ .Revision }}
+      - checkout
+      - run:
+          name: Temporarily rename TS files
+          command: |
+            mv src/contract-artifacts.ts src/contract-artifacts._ts
+            mv src/contract-deployed-artifacts.ts src/contract-deployed-artifacts._ts
+          working_directory: packages/contracts
+      - run:
+          name: Lint
+          command: yarn lint:check
+          working_directory: packages/contracts
+      - run:
+          name: Rename TS files back
+          command: |
+            mv src/contract-artifacts._ts src/contract-artifacts.ts
+            mv src/contract-deployed-artifacts._ts src/contract-deployed-artifacts.ts
+          working_directory: packages/contracts
+      - run:
+          name: Slither
+          command: yarn test:slither
+          working_directory: packages/contracts
+      - run:
+          name: Test
+          command: yarn test:coverage
+          working_directory: packages/contracts
+
+  dtl-tests:
+    docker:
+      - image: mslipper/slither:latest
+    steps:
+      - restore_cache:
+          keys:
+            - v1-yarn-build-{{ .Revision }}
+      - checkout
+      - run:
+          name: Test
+          command: yarn test:coverage
+          working_directory: packages/data-transport-layer
+
+  geth-tests:
+    docker:
+      - image: mslipper/go-builder:latest
+    steps:
+      - checkout
+      - run:
+          name: Test
+          command: make test
+          working_directory: l2geth
+
+  depcheck:
+    docker:
+      - image: mslipper/slither:latest
+    steps:
+      - restore_cache:
+          keys:
+            - v1-yarn-build-{{ .Revision }}
+      - checkout
+      - run:
+          name: Check contracts
+          command: npx depcheck
```

Comment:
Either needs to be noted as a manual thing to modify when packages are added/removed or we should have a `depcheck` script be added to the `package.json` files for each of these packages and simply run as a global monorepo script.